src/com/facebook/buck/jvm/groovy/GroovyLibraryDescription.java

```diff
@@ -26,31 +28,35 @@
 import com.facebook.buck.core.util.immutables.BuckStyleImmutable;
 import com.facebook.buck.io.filesystem.ProjectFilesystem;
 import com.facebook.buck.jvm.core.JavaAbis;
+import com.facebook.buck.jvm.groovy.GroovyLibraryDescription.AbstractGroovyLibraryDescriptionArg;
 import com.facebook.buck.jvm.java.DefaultJavaLibraryRules;
 import com.facebook.buck.jvm.java.JavaBuckConfig;
 import com.facebook.buck.jvm.java.JavaLibraryDescription;
 import com.facebook.buck.jvm.java.JavacOptions;
 import com.facebook.buck.jvm.java.JavacOptionsFactory;
 import com.facebook.buck.jvm.java.abi.AbiGenerationMode;
 import com.facebook.buck.jvm.java.toolchain.JavacOptionsProvider;
+import com.google.common.collect.ImmutableCollection.Builder;
 import com.google.common.collect.ImmutableList;
 import java.util.Optional;
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
 
 public class GroovyLibraryDescription
-    implements DescriptionWithTargetGraph<GroovyLibraryDescriptionArg> {
+    implements DescriptionWithTargetGraph<GroovyLibraryDescriptionArg>,
+        ImplicitDepsInferringDescription<AbstractGroovyLibraryDescriptionArg> {
 
   private final ToolchainProvider toolchainProvider;
-  private final GroovyBuckConfig groovyBuckConfig;
   private final JavaBuckConfig javaBuckConfig;
+  private final GroovyConfiguredCompilerFactory compilerFactory;
 
   public GroovyLibraryDescription(
       ToolchainProvider toolchainProvider,
-      GroovyBuckConfig groovyBuckConfig,
+      @Nullable GroovyBuckConfig groovyBuckConfig,
```

Comment:
I'm not really following here, how it can be null?

Reply:
this need not be null, but in tests this called with a null groovy buck config which is why I added `@Nullable`

https://github.com/facebook/buck/blob/3f1c33f5a4fed73559c890383374c6dcc9d9a59e/test/com/facebook/buck/jvm/groovy/GroovyLibraryBuilder.java#L51

Reply:
providing a non null instance now.